qiskit/transpiler/passes/synthesis/linear_functions_synthesis.py

```diff
@@ -28,15 +27,11 @@ class LinearFunctionsSynthesis(HighLevelSynthesis):
     Under the hood, this runs the default high-level synthesis plugin for linear functions.
     """
 
+    @deprecate_func(
+        additional_msg="Instead, use the class ``HighLevelSynthesis``.",
```

Comment:
Do we do Sphinx cross-refs in these messages? If so, then this can be `` :class:`.HighLevelSynthesis` ``.

Reply:
Indeed, we can use them. It will result in the runtime warning having some extra noise, hence https://github.com/Qiskit/qiskit-terra/pull/9815. We could either land that or not care about it being in the runtime warning. I don't have much preference.